src/Microsoft.ML.Data/DataView/DataViewExtensions.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.ML.Data
+{
+    [BestFriend]
+    internal static class DataViewExtensions
+    {
+        /// <summary>
+        /// Get a row cursor. The <paramref name="colsNeeded"/> are the active columns.
+        /// The schema of the returned cursor will be the same as the schema of the IDataView, but getting
+        /// a getter for an inactive columns will throw.
+        /// </summary>
+        /// <param name="colsNeeded">The columns requested by this <see cref="RowCursor"/>, or as otherwise called, the active columns.
+        /// An empty collection indicates that no column is needed.</param>
+        /// <param name="dv">The <see cref="IDataView"/> containing the <paramref name="colsNeeded"/>.</param>
+        [BestFriend]
+        internal static RowCursor GetRowCursor(this IDataView dv, params Schema.Column[] colsNeeded)
```

Comment:
This class appears to already be internal. So, can we simplify this by just getting rid of the `[BestFriend]` on the methods, and making the methods public? That's what we do everywhere else I believe.

Reply:
This also answers my question on Eric's previous PR :)

---
In reply to: [247007778](https://github.com/dotnet/machinelearning/pull/1975#discussion_r247007778) [](ancestors = 247007778)